File: templates/templates/php/Tests/basic-http/AuthAdditionalTest.php
declare(strict_types=1);

namespace Speakeasy\BasicHttp\Tests;

use Speakeasy\BasicHttp\Tests\CommonHelpers;
use Speakeasy\BasicHttp\Models\Components\Security;
use Speakeasy\BasicHttp\Models\Operations\BasicAuthOptionalSecurity;
use PHPUnit\Framework\TestCase;

final class AuthAdditionalTest extends TestCase
{
    public function testBasicAuthFlattenedGlobal(): void
    {
        CommonHelpers::recordTest('auth-basic-auth-flattened-global');

        $sdk = \Speakeasy\BasicHttp\SDK::builder()
            ->setSecurity(new Security(
                username: 'testUser',
                password: 'testPass'
            ))
            ->build();

        $response = $sdk->auth->basicAuthGlobal();
        $this->assertNotNull($response);
        $this->assertEquals(200, $response->httpMeta->response->getStatusCode());
        $this->assertNotNull($response->basicAuthResponse);
        $this->assertTrue($response->basicAuthResponse->authenticated);
        $this->assertEquals('testUser', $response->basicAuthResponse->user);
    }

    public function testBasicAuthFlattenedHoisted(): void
    {
        CommonHelpers::recordTest('auth-basic-auth-flattened-hoisted');

        $sdk = \Speakeasy\BasicHttp\SDK::builder()
            ->setSecurity(new Security(
                username: 'testUser',
                password: 'testPass'
            ))
            ->build();

        $response = $sdk->auth->basicAuthHoisted();
        $this->assertNotNull($response);
        $this->assertEquals(200, $response->httpMeta->response->getStatusCode());
        $this->assertNotNull($response->basicAuthResponse);
        $this->assertTrue($response->basicAuthResponse->authenticated);
        $this->assertEquals('testUser', $response->basicAuthResponse->user);
    }

    public function testBasicAuthOperationOptional(): void
    {
        // TODO: empty BasicAuthOptionalSecurity() still sends Authorization: Basic
        // header with empty credentials due to parseBasicAuthScheme not checking for empty fields.
        // Skipping negative test until PHP security util is fixed.
        $this->markTestSkipped('PHP security util does not yet reject empty basic-auth credentials');
    }
}

File: templates/templates/ruby/tests/basic-http/auth_additional_test.rb
# frozen_string_literal: true
# typed: true

require_relative "../lib/openapi"
require_relative "common_helper_test"

require "minitest/autorun"
require "minitest/focus"

class TestBasicAuthOptional < Minitest::Test
  def test_basic_auth_operation_optional
    record_test("auth-basic-auth-operation-optional")

    sdk = OpenApiSDK::SDK.new(
      security: OpenApiSDK::Models::Components::Security.new(
        username: "wrongUser",
        password: "wrongPass"
      )
    )

    res = sdk.auth.basic_auth_optional(
      security: OpenApiSDK::Models::Operations::BasicAuthOptionalSecurity.new(
        username: "testUser",
        password: "testPass"
      )
    )

    refute_nil(res)
    assert_equal(200, res.status_code)
    refute_nil(res.basic_auth_response)
    assert(res.basic_auth_response.authenticated)
    assert_equal("testUser", res.basic_auth_response.user)


    sdk2 = OpenApiSDK::SDK.new(
      security: OpenApiSDK::Models::Components::Security.new(
        username: "testUser",
        password: "testPass"
      )
    )

    assert_raises do
      sdk2.auth.basic_auth_optional(
        security: OpenApiSDK::Models::Operations::BasicAuthOptionalSecurity.new
      )
    end
  end
end
